### ISSUE TITLE
Paste with imports should not add imports when pasting into a string

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#59464]
### 🔎 Search Terms

- Paste with imports

### 🕗 Version & Regression Information

5.6.0-dev.20240729

### 💻 Code

For a project

```ts
// abc.ts
export const abc =123;

console.log(abc);
```

Copy and paste `abc` into a file:

```ts
const str = "|"
```

Where `|` is the cursor position 

### 🙁 Actual behavior

An import is added:

```ts
import { abc } from "./abc";

const a = 'abc'
```

### 🙂 Expected behavior

No import added

**Comment by mjbvz** [comment]
Other cases that are also broken:

- Paste into comment: `// |` or `/**  | */`
- Paste into regex: `/x|/g`
- Pasting into JSX body: `<b>|</b>`
- Past into JSX attribute: `<b className="|"></b>`

**Comment by IllusionMH** [comment]
> Pasting into JSX body: `<b>|</b>`

But what if you paste components and expressions? They might require imports

**Comment by mjbvz** [comment]
Yes for any of these cases, we also have to consider what the file will look like post paste. So for the string case (`'|'`), even though imports should not be added when pasting `abc`, they should be added when pasting `'+ abc + '` for example

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
